### PR TITLE
Skip FoR decompression if min and scalar are 0

### PR DESCRIPTION
**Pull request by robert3005**

### CODE REVIEW

File: encodings/fastlanes/src/for/compress.rs [hunk 1]
```diff
@@ -116,10 +116,14 @@ pub fn decompress(array: FoRArray) -> VortexResult<PrimitiveArray> {
                 .as_primitive()
                 .typed_value::<$T>()
                 .ok_or_else(|| vortex_err!("expected reference to be non-null"))?;
-            PrimitiveArray::from_vec(
-                decompress_primitive(encoded.into_maybe_null_slice::<$T>(), min, shift),
-                validity,
-            )
+            if min == 0 && shift == 0 {
+                encoded
```
Comment (lwwmanning) [inline encodings/fastlanes/src/for/compress.rs:120]: do we need to cast? (real question)

Comment (robert3005) [reply encodings/fastlanes/src/for/compress.rs:120]: In this context we know that min is T so I think compiler will cast 0 to T implicitly

Comment (lwwmanning) [reply encodings/fastlanes/src/for/compress.rs:120]: oh no, I mean cast `encoded`... is it the right dtype/ptype? since FoR converts to unsigned

Comment (robert3005) [reply encodings/fastlanes/src/for/compress.rs:120]: The cast happens above when we extract the encoded array

**Review by lwwmanning — APPROVED** [review]

**Pull request merged**